newrelic/hooks/mlmodel_openai.py

```diff
@@ -24,14 +24,53 @@
 from newrelic.core.config import global_settings
 
 
+def openai_error_attributes(exception, request_args):
+    # 'id' attribute will always be 'None' in this case.
+    # 'completion_id' is generated at the completion of
+    # the request and all of these error types occur
+    # before the response is generated.
+    #
+    api_key_LFD = None
+    if openai.api_key:
+        api_key_LFD = f"sk-{openai.api_key[-4:]}"
+
+    error_code = "None"
+    status_code = getattr(exception, "http_status", "None")
+    if status_code and status_code >= 400 and status_code < 600:
+        error_code = status_code
+
+    error_attributes = {
+        "id": "None",
+        "api_key_last_four_digits": api_key_LFD,
+        "request.model": request_args.get("model") or request_args.get("engine", "None"),
+        "temperature": request_args.get("temperature", "None"),
+        "max_tokens": request_args.get("max_tokens", "None"),
+        "vendor": "openAI",
+        "ingest_source": "Python",
+        "organization": getattr(exception, "organization", "None"),
+        "number_of_messages": len(request_args.get("messages", [])),
+        "status_code": status_code,
+        "error_message": getattr(exception, "_message", "None"),
+        "error_type": exception.__class__.__name__ or "None",
+        "error_code": error_code,
+        "error_param": getattr(exception, "param", "None"),
+    }
+    return error_attributes
+
+
 def wrap_embedding_create(wrapped, instance, args, kwargs):
     transaction = current_transaction()
     if not transaction:
         return wrapped(*args, **kwargs)
 
     ft_name = callable_name(wrapped)
     with FunctionTrace(ft_name) as ft:
-        response = wrapped(*args, **kwargs)
+        try:
+            response = wrapped(*args, **kwargs)
+        except Exception as exc:
+            error_attributes = openai_error_attributes(exc, kwargs)
+            transaction.notice_error(attributes=error_attributes)
```

Comment:
We need notice_error to pick up some additional attributes that aren't picked up in transaction's notice_error. Specifically [these](https://github.com/newrelic/newrelic-python-agent/blob/main/newrelic/api/time_trace.py#L356-L358):
```
        # Add error details as agent attributes to span event.
        self._add_agent_attribute("error.class", fullname)
        self._add_agent_attribute("error.message", message)
        self._add_agent_attribute("error.expected", is_expected)
```
Luckily function_trace inherits from time_trace and picks these up for us.
```suggestion
            ft.notice_error(attributes=error_attributes)
```

Along with this it looks like we need to make a change to the notice_error there to optionally take in a message argument to set the message.